src/Orchard/Environment/OrchardStarter.cs

```diff
@@ -67,6 +67,7 @@ public static class OrchardStarter {
             builder.RegisterType<ViewsBackgroundCompilation>().As<IViewsBackgroundCompilation>().SingleInstance();
             builder.RegisterType<DefaultExceptionPolicy>().As<IExceptionPolicy>().SingleInstance();
             builder.RegisterType<DefaultCriticalErrorProvider>().As<ICriticalErrorProvider>().SingleInstance();
+            builder.RegisterType<ThreadProvider>().As<IThreadProvider>().SingleInstance();

```

Comment:
Technically it is correct, since the service does not hold on to any data - it merely provides access to the static `Thread.CurrentThread` object:

```C#
public class ThreadProvider : IThreadProvider {
    public int GetCurrentThreadId() {
        return Thread.CurrentThread.ManagedThreadId;
    }
}
```

If it's ethical, that's up for debate I suppose.